api/base/utils.py

```diff
@@ -102,3 +103,32 @@ def waterbutler_url_for(request_type, provider, path, node_id, token, obj_args=N
 
     url.args.update(query)
     return url.url
+
+
+def soft_get(something, key):
```

Comment:
Should this (and deep_get) just go into website.utils? They seem generally useful.